integration/Substra/plot_substrafl_flamby.py

```diff
@@ -0,0 +1,427 @@
+    DatasetSpec(
+        name="FLamby",
+        type="torchDataset",
+        data_opener=assets_directory / "dataset" / "opener_pooled_test.py",
+        description=assets_directory / "dataset" / "description.md",
+        permissions=permissions,
+        logs_permission=permissions,
+    )
+)
+assert test_dataset_key, "Missing data manager key"
+
+data_sample = DataSampleSpec(
+    data_manager_keys=[test_dataset_key],
+    test_only=True,
+    path=pathlib.Path.cwd() / "empty",
+)
+test_datasample_key = client.add_data_sample(
+    data_sample,
+    local=True,
+)
+
+# Create the Test Data Node (or testing task) and save it in a list
+test_data_node = TestDataNode(
+    organization_id=ALGO_ORG_ID,
+    data_manager_key=test_dataset_key,
+    test_data_sample_keys=[test_datasample_key],
+    metric_keys=[metric_key],
+)
+test_data_nodes.append(test_data_node)
+
+# %%
+# Machine Learning specification
+# ******************************
+
+from flamby.datasets.fed_tcga_brca import (
+    Baseline as baseline_model,
+    BaselineLoss as baseline_loss,
+    Optimizer as optimizer,
+    BATCH_SIZE as batch_size,
+    LR as ixi_lr,
+)
+
+
+model = baseline_model()
+optimizer = optimizer(model.parameters(), lr=ixi_lr)
+criterion = baseline_loss()
+
+# %%
+# Substrafl imports
+# ==================
+
+from typing import Any
+
+from substrafl.algorithms.pytorch import TorchFedAvgAlgo
+from substrafl.dependency import Dependency
+from substrafl.strategies import FedAvg
+from substrafl.nodes import AggregationNode
+from substrafl.evaluation_strategy import EvaluationStrategy
+from substrafl.index_generator import NpIndexGenerator
+from substrafl.experiment import execute_experiment
+
+import torch
+
+# %%
+# Substrafl algo definition
+# ==========================
+#
+# To instantiate a Substrafl :ref:`substrafl_doc/api/algorithms:Torch Algorithms`, you need to define a torch Dataset
+# with a specific `__init__` signature, that must contain (self, x, y, is_inference). This torch Dataset is useful to
+# preprocess your data on the `__getitem__` function.
+# The `__getitem__` function is expected to return x and y if is_inference is False, else x.
+# This behavior can be changed by re-writing the `_local_train` or `predict` methods.
+#
+# This dataset is passed **as a class** to the :ref:`substrafl_doc/api/algorithms:Torch Algorithms`.
+# Indeed, this torch Dataset will be instantiated within the algorithm, using the opener functions as x and y
+# parameters.
+#
+# The index generator will be used a the batch sampler of the dataset, in order to save the state of the seen samples
+# during the training, as Federated Algorithms have a notion of `num_updates`, which forced the batch sampler of the
+# dataset to be stateful.
+
+# Number of model update between each FL strategy aggregation.
+NUM_UPDATES = 16
+
+index_generator = NpIndexGenerator(
+    batch_size=batch_size,
+    num_updates=NUM_UPDATES,
+)
+
+
+class TorchDataset:
```

Comment:
Use inheritance instead of this please

Reply:
I changed the interaction to make this part more readable. Now, the opener does not return the FLamby dataset but a dict containing the argument to instantiate it. I like it better this way thanks for insisting on this one 😄 

The opener for testing still instantiate a dataset to access to the y values though...